lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -891,7 +891,9 @@ void HdVP2Mesh::Sync(
 
         auto addRequiredPrimvars = [&](const SdfPath& materialId) {
             TfTokenVector requiredPrimvars;
-            if (!_GetMaterialPrimvars(renderIndex, materialId, requiredPrimvars)) {
+            // Only load textures when texture mode is selected
+            if (!_GetMaterialPrimvars(renderIndex, materialId, requiredPrimvars)
+                || !(reprToken == HdReprTokens->smoothHull)) {
```

Comment:
This looks wrong. If your intention here is to choose sFallbackShaderPrimvars for untextured mode then you need to have something like: || (reprToken == HdReprTokens->smoothHullUntextured)

Reply:
Also, in case where hasColor is true below, we don't use fallback shader. So I guess this logic should be consistent